build/source/engine/HDS.f90

```diff
@@ -0,0 +1,321 @@
+module HDS
+    USE nrtype
+    ! physical constants
+    USE multiconst,only:LH_vap,  & ! latent heat of vaporization   (J kg-1)
+                        iden_water ! intrinsic density of water    (kg m-3)
+    
+    contains
+    subroutine init_summa_HDS(pondVolFrac, depressionDepth, depressionAreaFrac, totalArea, p, pondVol, pondArea, conArea, vMin)
+        ! initialize HDS pothole storage model states for SUMMA
+        implicit none
+        !subroutine arguments
+        real(rkind),  intent(inout)  :: pondVolFrac                    ! depression volume fraction [-]
+        real(rkind),  intent(in)     :: depressionDepth                ! depression depth [m]
+        real(rkind),  intent(in)     :: depressionAreaFrac             ! depression area fraction [-]
+        real(rkind),  intent(in)     :: totalArea                      ! basin (GRU) total area [m2]
+        real(rkind),  intent(in)     :: p                              ! shape of the slope profile [-]. Exponent for calculating the fractional wet area
+        real(rkind),  intent(out)    :: pondVol                        ! pond volume [m3]
+        real(rkind),  intent(out)    :: pondArea                       ! pond Area [m2]
+        real(rkind),  intent(out)    :: conArea                        ! fractional contributing area [-]
+        real(rkind),  intent(out)    :: vMin                           ! minimum pond volume [m3]
+
+        ! local variables
+        real(rkind)                  :: depressionArea                 ! depression area [m2]
+        real(rkind)                  :: depressionVol                  ! depression volume [m3]
+        !*****************************************
+        ! initialize HDS pothole storage variables
+        ! GRU level
```

Comment:
Consider updating to new refactored code standards that are currently being developed

Reply:
Example code is here:
https://github.com/CH-Earth/summa/blob/develop_sundials/build/source/engine/computFlux.f90#L242-L249